### PR TITLE
Add poultry meat

### PR DESCRIPTION
**Pull request by XygenSS**
#### Summary
Content "Add poultry meat"

#### Purpose of change

Bird meat is nutritionally distinct enough from meat from other land animals.
Also, fried chicken.

#### Describe the solution

Birds yield poultry meats. Can be smoked and dehydrated.
Bird meat was added to existing recipe requirement groups. To be revisited later on a followup PR.
8 new recipes that use poultry meat specifically.

A chunk of poultry is roughly "one portion," approx. 4oz or 100g for ease of usage in food recipes.
A scrap of poultry is 1/5th of a chunk, 20g, unlike scrap of meat which is 1/10th of a chunk.

Nutrition value is based on skin on, boneless, wild duck meat from USDA Database: https://fdc.nal.usda.gov/fdc-app.html#/food-details/174468/nutrients 

<details>
<summary>List of new recipes</summary>

- [x] Cooked/pepper/smoked/dehydrated chicken
- [x] Cooked/pepper scrap of chicken
- [x] Fried chicken
- [x] Buttermilk fried chicken (vinegar leg on the right)
- [x] Tandoori chicken
- [x] Chicken sandwich and wheat-free sandwich (which is a burger)
- [x] Coq au vin
- [x] Hianese chicken rice (Singaporean recipe, too similar to deluxe fried rice?)
- [x] Chicken parm

</details>

#### Describe alternatives you've considered

~~1. Lumping poultry meat and waterfowl meat together. This would mean that aquatic birds would simply yield more fat when butchered.~~ 
~~2. Basing the nutrition value on another animal or another cut.~~
3. Differentiating salmonella from food poisoning (in a future PR).

All poultry meat will be based on "wild" duck meat. It is close enough to domesticated chicken meat, which is what the survivor is most likely to farm/consume, and bridges the gap between lean birds and fatty waterfowl. Even though domesticated and farmed waterfowl like goose and duck approach 400kcal per portion, wild birds as the survivor is likely to hunt in the wild are not as calorically dense.

#### Testing

- [x] Verify butchering a chicken
- [x] Verify butchering a waterfowl
- [x] Smoke and dehydrate products
- [x] Verify existing food recipes
- [x] Test new food items and their recipes; recipe categories in crafting menus
- [x] Verify recipe books
- [x] General balance pass

#### Additional context

Todo in a future PR:
- Adjust butchering results of various birds to yield more chunks and less scraps (to a point).
- If needed, add more fat to butchering results of waterfowl.
- Remove chicken from recipes that don't make sense with it.
- Merge canned chicken and cooked chicken, and obsolete them.
- Stop putting French food in an Italian cookbook.

### CODE REVIEW

File: data/json/items/comestibles/carnivore.json [hunk 1]
```diff
@@ -1951,5 +1951,43 @@
...
+    "weight": "160 g",
+    "volume": "250 ml",
+    "price": 500,
+    "price_postapoc": 50,
+    "spoils_in": "1 day",
+    "calories": 240,
+    "//": "Scaled from 100g boneless chicken thigh according to USDA database"
+    "fun": -20,
+    "cooks_like": "meat_cooked",
+    "vitamins": [ [ "calcium", "9.6 mg" ], [ "iron", "0.96 mg" ] ],
```
Comment (ehughsbaird) [inline data/json/items/comestibles/carnivore.json:1970]: These must be integers
```suggestion
    "vitamins": [ [ "calcium", "9 mg 600 μg" ], [ "iron", "960 μg" ] ],
```

File: data/json/items/comestibles/carnivore.json [hunk 2]
```diff
@@ -1951,5 +1951,43 @@
...
+    "weight": "160 g",
+    "volume": "250 ml",
+    "price": 500,
+    "price_postapoc": 50,
+    "spoils_in": "1 day",
+    "calories": 594,
+    "//": "Scaled from 100g skin-on goose according to USDA database"
+    "fun": -20,
+    "cooks_like": "meat_cooked",
+    "vitamins": [ [ "calcium", "19.2 mg" ], [ "iron", "4 mg" ] ],
```
Comment (ehughsbaird) [inline data/json/items/comestibles/carnivore.json:1989]:
```suggestion
    "vitamins": [ [ "calcium", "19 mg 200 μg" ], [ "iron", "4 mg" ] ],
```

**Comment by XygenSS** [comment]
I messed up git-fu real bad and had to rebase.
Let's not do this ever again whoa

**Comment by XygenSS** [comment]
Time for testing, since I had none of the initial questions answered I expect I'll have to redo a lot of stuff, but here we go anyways.

**Comment by RedMisao** [comment]
Hello, I've been keeping an eye on this since it's probably the closest to someone implementing bird meat in mainline

Some suggestions for your bullet points (may overlap):
1. Yes, this is a good approach. It's simpler both for the game and for you to lump _all_ bird-sourced flesh into a single item, This will also be similar to how currently "meat" lumps everything that has flesh together, doesn't matter the species or the fine characteristics of the body you're getting the meat from. I'd base bird meat into either chicken for the sake of simplicity, or what is the most commonly consumed bird animal in New England
2. Same as 1.
3. Similar to 1., I'd just lump both in the same category. Maybe if there's any dishes that require the use of fatty bird meat _exclusively_ over normal bird meat, put a piece of fat in their recipe instead. For practical purposes, as the objective of adding bird meat is to have recipes using only that over generic meat, you have to weigh how practical it would be to have both kinds of bird meat
4. It depends on you. Maybe add a note in the bird meat specifying that fatty bird meat shouldn't be preserved like this
5. Same as 1.
6. Yes, the tl;dr story is that _Salmonella_ may as well be a comensal part of their microbiota, which is just a problem for humans eating them. There's a `POISON` flag that can be removed after cooking to emulate this (I'm not sure how this works, I may be mistaken about this), or if you want to do it the proper way add a Salmonellosis disease, which may be out of scope for the current PR
7. I'd go with "bird X" to be as encompassing as possible

Suggestions for recipes: chicken pie, chicken pot pie, chicken nuggets/tenders (!), roasted whole bird (dunno how you would handle this, maybe the entire bird corpse item?), skewered chicken

Also, you should also ask in the official discord for what more experienced people think about this

**Comment by XygenSS** [comment]
@RedMisao Thanks for the input.

As for additional recipes the PR is already pusing it in terms of string count, so they will have to come later. I am also planning a spices PR so it would have to come after that as well.

Since raw bird meat copies from raw meat it currently gives you the same sort of food poisoning. If salmonella is desired it will happen in the future for sure

I did post on discord a few times, I'll sit on merging bird meat types until someone higher than my paygrade can review it.

File: data/json/items/comestibles/carnivore.json [hunk 3]
```diff
@@ -236,7 +236,6 @@
     "name": { "str_sp": "human flesh" },
     "description": "A chunk of raw meat, butchered from a human body.",
     "material": [ "hflesh" ],
-    "vitamins": [ [ "calcium", 2 ], [ "iron", 12 ], [ "human_flesh_vitamin", 100 ] ],
```
Comment (Procyonae) [inline data/json/items/comestibles/carnivore.json:239]: Is this an intentional change?

File: data/json/items/comestibles/carnivore.json [hunk 4]
```diff
@@ -388,7 +387,6 @@
     "looks_like": "meat_scrap",
     "name": { "str": "scrap of human flesh", "str_pl": "scraps of human flesh" },
     "description": "A tiny scrap of raw human flesh.",
-    "vitamins": [ [ "calcium", 0 ], [ "iron", 1 ], [ "human_flesh_vitamin", 10 ] ],
```
Comment (Procyonae) [inline data/json/items/comestibles/carnivore.json:391]: Ditto

File: data/json/items/comestibles/carnivore.json (hunk 3 above)
Comment (XygenSS) [reply data/json/items/comestibles/carnivore.json:239]: No clue how that got there, good catch

Comment (Procyonae) [reply data/json/items/comestibles/carnivore.json:239]: There's a few other oddities to do with cannibal stuff you probably want to check the files changes tab

Comment (XygenSS) [reply data/json/items/comestibles/carnivore.json:239]: alright

**Review by GuardianDll — APPROVED** [review]
I checked it and didn't spot any obvious issues; i much prefer if you'd move bird_tiny, bird_small and bird_large harvest lists back where they were, instead of moving it to the end, to preserve history of changes
As long as all test errors would be resolved, i see no issues in merging it

File: data/json/items/comestibles/meat_dishes.json [hunk 5]
```diff
@@ -1554,6 +1554,15 @@
     "volume": "25 ml",
     "flags": [ "EDIBLE_FROZEN" ]
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "dry_poultry",
+    "name": { "str_sp": "dehydrated chicken" },
```
Comment (Karol1223) [inline data/json/items/comestibles/meat_dishes.json:1560]:
```suggestion
    "name": { "str_sp": "dehydrated poultry" },
```

File: data/json/items/comestibles/meat_dishes.json [hunk 6]
```diff
@@ -1565,6 +1574,17 @@
     "price_postapoc": 300,
     "fun": 2
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "rehydrated_poultry",
+    "name": { "str": "rehydrated chicken" },
```
Comment (Karol1223) [inline data/json/items/comestibles/meat_dishes.json:1580]:
```suggestion
    "name": { "str": "rehydrated poultry" },
```

File: data/json/items/comestibles/meat_dishes.json [hunk 7]
```diff
@@ -1565,6 +1574,17 @@
...
+    "copy-from": "poultry_cooked",
+    "color": "light_red",
+    "description": "Reconstituted bird meat, which are much more enjoyable to eat now that they have been rehydrated.",
```
Comment (Karol1223) [inline data/json/items/comestibles/meat_dishes.json:1583]:
```suggestion
    "description": "Reconstituted bird meat, which is much more enjoyable to eat now that it has been rehydrated.",
```

File: data/json/items/comestibles/carnivore.json [hunk 8]
```diff
@@ -2075,5 +2075,127 @@
...
+    "parasites": 0,
+    "calories": 365,
+    "flags": [ "EATEN_HOT" ]
+  },
+  {
+    "id": "poultry_scrap_pepper",
+    "copy-from": "meat_scrap",
+    "type": "COMESTIBLE",
+    "name": { "str": "cooked pepper scrap of chicken", "str_pl": "cooked pepper scraps of chicken" },
+    "description": "A cooked scrap of bird that has been properly seasoned with pepper and salt.  Yet small, they have a good taste.",
```
Comment (Karol1223) [inline data/json/items/comestibles/carnivore.json:2169]:
```suggestion
    "description": "A cooked scrap of bird that has been properly seasoned with pepper and salt.  Small, yet it has a good taste.",
```

File: data/json/requirements/cooking_components.json [hunk 9]
```diff
@@ -667,7 +688,7 @@
         [ "oats", 1 ],
         [ "oatmeal", 1 ]
       ],
-      [ [ "meat_red", 1, "LIST" ], [ "dry_meat", 1 ], [ "can_chicken", 1 ] ]
+      [ [ "meat_red", 1, "LIST" ], [ "dry_meat", 1 ], [ "poultry_raw_any", 1, "LIST" ], [ "dry_poultry", 1 ] ]
```
Comment (Karol1223) [inline data/json/requirements/cooking_components.json:691]: This can no longer use canned chicken. Is that intended?

**Review by Karol1223 — CHANGES_REQUESTED** [review]

**Comment by Karol1223** [comment]
> I checked it and didn't spot any obvious issues; i much prefer if you'd move bird_tiny, bird_small and bird_large harvest lists back where they were, instead of moving it to the end, to preserve history of changes As long as all test errors would be resolved, i see no issues in merging it

Also +1 on that front. Avoid moving file contents without a reason. Put the harvestlists back where they were, add the waterfowl ones under them instead of at the end of the file so they're in one spot.

File: data/json/items/comestibles/carnivore.json [hunk 10]
```diff
@@ -2075,5 +2075,127 @@
...
+  },
+  {
+    "id": "poultry",
+    "copy-from": "meat",
+    "type": "COMESTIBLE",
+    "name": { "str": "chunk of poultry meat", "str_pl": "chunks of poultry meat" },
+    "description": "A chunk of raw meat from a lean bird.  You could eat it raw, but it carries a risk of bacterial infection.",
+    "weight": "160 g",
+    "calories": 240,
+    "//": "Scaled from 100g boneless chicken thigh according to USDA database",
```
Comment (Maleclypse) [inline data/json/items/comestibles/carnivore.json:2087]: Is this 100g of meat also 250ml in volume since it's copied the volume of the existing meat?  Please confirm

Comment (Maleclypse) [reply data/json/items/comestibles/carnivore.json:2087]: <https://www.aqua-calc.com/page/density-table/substance/beef-blank-steak-coma-and-blank-ns-blank-as-blank-to-blank-cooking-blank-method-coma-and-blank-lean-blank-and-blank-fat-blank-eaten-blank--op-boneless-coma-and-blank-cooked-cp->  
I'm seeing 1.04 g/ml for both beef and chicken.  So you should have 260g of poultry and our meat item should also be 260g?

**Review by Maleclypse — CHANGES_REQUESTED** [review]
I think there are some areas that need further detail or explanation of sources and this is also leading to some questions about items that are copied from in this PR.  I like where it's going though!

Looks like calories should be in a ratio of 78% of red meat?

File: data/json/items/comestibles/meat_dishes.json [hunk 11]
```diff
@@ -1565,6 +1574,17 @@
     "price_postapoc": 300,
     "fun": 2
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "rehydrated_poultry",
+    "name": { "str": "rehydrated poultry" },
```
Comment (Karol1223) [inline data/json/items/comestibles/meat_dishes.json:1580]:
```suggestion
    "name": { "str_sp": "rehydrated poultry" },
```

**Review by Karol1223 — CHANGES_REQUESTED** [review]

File: data/json/items/comestibles/carnivore.json (hunk 10 above)
Comment (XygenSS) [reply data/json/items/comestibles/carnivore.json:2087]: What exactly is our meat based on, as a reference?

Comment (Maleclypse) [reply data/json/items/comestibles/carnivore.json:2087]: Hopefully somewhere in this PR is an answer.  This is the kind of detail we should have in comments on baseline items like meat.  https://github.com/CleverRaven/Cataclysm-DDA/pull/27289

**Comment by XygenSS** [comment]
Here's what I think I'll gonna do.
Instead of anchoring volume or weight on existing chunk of meat, since poultry meat is going to be predominantly used in recipes I think it would save a lot of headache if a "chunk of poultry" is standardized to be roughly one portion in cooking.

Serving size per person for chicken breast (breast because it is boneless) is said to be 3~5 ounces. Let's say 4 ounces, 113g. At 1.04g/ml that will be 109ml. Anchoring the volume at 100ml means we have 104g of poultry per portion.

Using the USDA Database and basing poultry on boneless skin-on chicken thighs (https://fdc.nal.usda.gov/fdc-app.html#/food-details/172385/nutrients) and fatty poultry on boneless skin-on goose (https://fdc.nal.usda.gov/fdc-app.html#/food-details/174470/nutrients) should mean that ingame poultry is 230kcal and fatty poultry is 386kcal.

However, if we were to base fatty poultry on duck, we have a problem: "domesticated" duck is roughly 400+kcal/100g (https://fdc.nal.usda.gov/fdc-app.html#/food-details/172408/nutrients), while "wild" duck is only 200kcal/100g (https://fdc.nal.usda.gov/fdc-app.html#/food-details/174468/nutrients). If we were to accept the "wild" duck nutrition as more realistic this probably means splitting regular and fatty poultry meat is unnecessary and should be redone.

This was one of the unanswered questions on the original PR main body which I kind of glanced over to get things moving. If we need a solid foundation on accurate nutrition data going forward it would be best to clarify this right now.

TLDR;
- one chunk of poultry will be 100ml
- Fatty poultry will not be needed and can be merged with regular poultry if we consider the meats of random wild waterfowl ingame is nutritionally similar to "wild duck" according to USDA
- Otherwise it can be assumed to be similar to domesticated duck or goose & will be split as two different kinds of meat

**Comment by XygenSS** [comment]
@Maleclypse would like to hear what you think.

**Comment by Maleclypse** [comment]
Fatty poultry will not be needed and can be merged with regular poultry if we consider the meats of random wild waterfowl ingame is nutritionally similar to "wild duck" according to USDA

Let's do this.  Even if the survivor were to domesticate poultry lack of access to modern farming techniques would likely make those poultry similar to wild poultyry.

**Comment by Karol1223** [comment]
> Leaving the "waterfowl bird" harvest groups in, in case someone might revisit it later and add more fat into the butchering results.

Please don't. I understand why, but with this you're just adding a bunch more work for people who are gonna be working with birds next (me) as well as an extra currently useless abstract chick.

Don't assume someone is gonna pick up where you left off. People rarely do that. If they want to do the thing, they'll do it by themselves.

**Comment by XygenSS** [comment]
@Karol1223 Noted, all changes removed

**Review by Maleclypse — CHANGES_REQUESTED** [review]
Sorry I missed these two missing prices before but once they have prices I think this is good to merge.  Thank you!

**Comment by XygenSS** [comment]
@Maleclypse prices are in, thanks for catching it!

**Comment by Karol1223** [comment]
> @Karol1223 Noted, all changes removed

They're literally still there chief

**Comment by XygenSS** [comment]
@Karol1223 derp, appreciate the catch

File: data/json/requirements/cooking_components.json [hunk 12]
```diff
@@ -95,15 +95,12 @@
...
         [ "egg_grouse", 1 ],
-        [ "egg_crow", 3 ],
-        [ "egg_raven", 3 ],
-        [ "egg_bluejay", 6 ],
-        [ "egg_cardinal", 12 ],
-        [ "egg_robin", 7 ],
-        [ "egg_pigeon", 3 ],
-        [ "egg_sparrow", 20 ],
-        [ "egg_waxwing", 20 ],
-        [ "egg_chickadee", 50 ],
```
Comment (Karol1223) [inline data/json/requirements/cooking_components.json:106]: You're reverting my changes from my bird audit here

**Review by Karol1223 — CHANGES_REQUESTED** [review]

File: data/json/requirements/cooking_components.json (hunk 12 above)
Comment (XygenSS) [reply data/json/requirements/cooking_components.json:106]: I never got notified for merge conflicts and never deleted these myself?

I'll look into it. I hate git

**Review by Karol1223 — APPROVED** [review]
Can't find anything that seems wrong. I could complain about some item naming but honestly, it's good enough in my eyes. The mapgen tests are just throwing tantrums yet again.

**Comment by XygenSS** [comment]
ITS OVER
